src/app/clusters/scenes/SceneTable.h

```diff
@@ -0,0 +1,293 @@
+        // Identifies group within the scope of the given fabric
+        SceneGroupID mGroupId = kGlobalGroupSceneId;
+        SceneId mSceneId      = kUndefinedSceneId;
+
+        SceneStorageId() = default;
+        SceneStorageId(EndpointId endpoint, SceneId id, SceneGroupID groupId = kGlobalGroupSceneId) :
+            mEndpointId(endpoint), mGroupId(groupId), mSceneId(id)
+        {}
+
+        void Clear()
+        {
+            mEndpointId = kInvalidEndpointId;
+            mGroupId    = kGlobalGroupSceneId;
+            mSceneId    = kUndefinedSceneId;
+        }
+
+        bool operator==(const SceneStorageId & other)
+        {
+            return (this->mEndpointId == other.mEndpointId && this->mGroupId == other.mGroupId && this->mSceneId == other.mSceneId);
+        }
+    };
+
+    /// @brief struct used to store data held in a scene
+    struct SceneData
+    {
+        char mName[kSceneNameMax]                       = { 0 };
+        size_t mNameLength                              = 0;
+        SceneTransitionTime mSceneTransitionTimeSeconds = 0;
+        EFStype mExtensionFieldSets;
+        TransitionTime100ms mTransitionTime100ms = 0;
+
+        SceneData(const CharSpan & sceneName = CharSpan(), SceneTransitionTime time = 0, TransitionTime100ms time100ms = 0) :
+            mSceneTransitionTimeSeconds(time), mTransitionTime100ms(time100ms)
+        {
+            this->SetName(sceneName);
+        }
+        SceneData(EFStype fields, const CharSpan & sceneName = CharSpan(), SceneTransitionTime time = 0,
+                  TransitionTime100ms time100ms = 0) :
+            mSceneTransitionTimeSeconds(time),
+            mTransitionTime100ms(time100ms)
+        {
+            this->SetName(sceneName);
+            mExtensionFieldSets = fields;
+        }
+        SceneData(const SceneData & other) :
+            mSceneTransitionTimeSeconds(other.mSceneTransitionTimeSeconds), mTransitionTime100ms(other.mTransitionTime100ms)
+        {
+            this->SetName(CharSpan(other.mName, other.mNameLength));
+            mExtensionFieldSets = other.mExtensionFieldSets;
+        }
+        ~SceneData(){};
+
+        void SetName(const CharSpan & sceneName)
+        {
+            if (nullptr == sceneName.data())
+            {
+                mName[0]    = 0;
+                mNameLength = 0;
+            }
+            else
+            {
+                size_t maxChars = std::min(sceneName.size(), kSceneNameMax);
+                memcpy(mName, sceneName.data(), maxChars);
+                mNameLength = maxChars;
+            }
+        }
+
+        void Clear()
+        {
+            this->SetName(CharSpan());
+            mSceneTransitionTimeSeconds = 0;
+            mTransitionTime100ms        = 0;
+            mExtensionFieldSets.Clear();
+        }
+
+        bool operator==(const SceneData & other)
+        {
+            return (!memcmp(this->mName, other.mName, this->mNameLength) &&

```

Comment:
This can compare garbage bytes if the other data's mNameLength is different.  We should be checking that the lengths match.